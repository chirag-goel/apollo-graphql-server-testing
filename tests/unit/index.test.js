const {
    ApolloServerBase
} = require('apollo-server-core');
const typeDefs = require('@src/typeDefs.js');
const resolvers = require('@src/resolvers.js');
const {
    createTestClient
} = require('apollo-server-testing');

// Schema testing
describe('Schema Testing With Mocked Schema', () => {
    const mocks = {
        Float: () => 22.1,
        Int: () => 6,
        String: () => 'Hello',
    };

    const myTestServer = new ApolloServerBase({
        mocks,
        resolvers,
        typeDefs,
    });
    it('QUERY_hello', async () => {
        const query = '{ hello }';
        const client = createTestClient(myTestServer);
        const res = await client.query({
            query
        });
        expect(res.data.hello).toEqual('Hello');
    });

    it('QUERY_hello_withSnapshot', async () => {
        const query = '{ hello }';
        const client = createTestClient(myTestServer);
        const res = await client.query({
            query
        });
        expect(res.data.hello).toMatchSnapshot();
    });
});

// Resolver testing
describe('Resolver Testing Without Mock', () => {

    const myTestServer = new ApolloServerBase({
        resolvers,
        typeDefs,
    });

    it('MUTATION_greet_withSnapshot', async () => {
        const mutation = 'mutation { greet(name: "Chirag") }';
        const client = createTestClient(myTestServer);
        const res = await client.mutate({
            mutation
        });
        expect(res.data.greet).toMatchSnapshot();
    });
});