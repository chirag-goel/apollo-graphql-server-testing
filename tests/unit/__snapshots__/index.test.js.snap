// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resolver Testing Without Mock MUTATION_greet_withSnapshot 1`] = `"Welcome Chirag!"`;

exports[`Schema Testing With Mocked Schema QUERY_hello_withSnapshot 1`] = `"Hello"`;
